BlinkStickDotNet/Colors/HSLColor.cs

```diff
@@ -0,0 +1,174 @@
+using System;
+
+namespace BlinkStickDotNet.Colors
+{
+    /// <summary>
+    /// Defines an HSL color.
+    /// </summary>
+    /// <remarks>Check http://en.wikipedia.org/wiki/HSL_color_space for more info.</remarks>
+    public struct HSLColor
+    {
+        private double _hue;
+        private double _saturation;
+        private double _lightness;
```

Comment:
If this is to be mutable, it should be a class not a struct.

https://blogs.msdn.microsoft.com/ericlippert/2008/05/14/mutating-readonly-structs/

My sense (which follows the BCL colour types) is to make HslColor and RbgColor immutable structs.